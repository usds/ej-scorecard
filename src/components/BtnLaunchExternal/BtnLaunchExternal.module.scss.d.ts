declare namespace BtnLaunchExternalNamespace {
  export interface IBtnLaunchExternalScss {
    btnLaunch: string;
    icon: string;
  }
}

declare const BtnLaunchExternalScssModule: BtnLaunchExternalNamespace.IBtnLaunchExternalScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: BtnLaunchExternalNamespace.IBtnLaunchExternalScss;
};

export = BtnLaunchExternalScssModule;
