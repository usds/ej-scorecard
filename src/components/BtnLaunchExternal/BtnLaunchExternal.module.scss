@use "uswds-core" as *;

.btnLaunch {
    @include u-padding-bottom(2);

    .icon {
        margin-left: 5px;
        margin-bottom: -5px;
        width: 20%;
        filter: invert(100%) sepia(18%) saturate(129%) hue-rotate(237deg) brightness(113%) contrast(100%);
    }
}
