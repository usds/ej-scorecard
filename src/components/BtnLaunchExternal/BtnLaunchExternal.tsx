import React from 'react';
import { Button } from '@trussworks/react-uswds';
import { BtnLaunchExternalProps } from '@/types';

import * as styles from './BtnLaunchExternal.module.scss';

// @ts-ignore
// eslint-disable-next-line import/no-absolute-path
import launchIcon from '/node_modules/@uswds/uswds/dist/img/usa-icons/launch.svg';

const BtnLaunchExternal: React.FC<BtnLaunchExternalProps> = ({
  href,
  buttonText,
  showlaunchIcon = true,
}) => {
  return (
    <a
      className={`flex-align-center`}
      href={href}
      target={`_blank`}
      rel="noreferrer"
    >
      <Button type="button" className={styles.btnLaunch}>
        <div>
          {buttonText}
          {showlaunchIcon ? (
            <img className={styles.icon} src={launchIcon} alt={`launch icon`} />
          ) : (
            <></>
          )}
        </div>
      </Button>
    </a>
  );
};

export default BtnLaunchExternal;
