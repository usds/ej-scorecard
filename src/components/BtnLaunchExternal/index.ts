import BtnLaunchExternal from './BtnLaunchExternal';
export default BtnLaunchExternal;
