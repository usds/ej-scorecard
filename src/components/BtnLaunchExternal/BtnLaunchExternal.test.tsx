import createComponentWithIntl from '@/test/testHelpers';

import BtnLaunchExternal from './BtnLaunchExternal';

describe(`rendering of BtnLaunchExternal Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <BtnLaunchExternal
        href="www.google.com"
        buttonText="test"
        showlaunchIcon={true}
      />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
