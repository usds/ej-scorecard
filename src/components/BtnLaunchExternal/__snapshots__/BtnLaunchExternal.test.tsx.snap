// Vitest Snapshot v1

exports[`rendering of BtnLaunchExternal Component > checks if component renders 1`] = `
<a
  className="flex-align-center"
  href="www.google.com"
  rel="noreferrer"
  target="_blank"
>
  <button
    className="usa-button"
    data-testid="button"
    type="button"
  >
    <div>
      test
      <img
        alt="launch icon"
        src="/node_modules/.pnpm/@uswds+uswds@3.3.0/node_modules/@uswds/uswds/dist/img/usa-icons/launch.svg"
      />
    </div>
  </button>
</a>
`;
